red/reporter.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2015 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var Log = require("./log");
+var TEN_SECONDS_MILLIS = 10000;
+
+var reporter = {
+    init: function(settings) {
+        if(settings) {
+            // TODO settings might be used in the future for configuring runtime reports
+        }
+        startInterval(TEN_SECONDS_MILLIS);
+    }
```

Comment:
This should be more closely tied to the life-cycle of node-red.

In addition to the `init` function, this module should have a pair of `start` and `stop` functions that can be used to start/stop the interval, rather than leave it running indefinitely.

There is a pair of `start` and `stop` in `red.js` that currently delegate straight through to `server.start`/`server.stop`. They should be replaced with functions that first call `reporter.start`/`reporter.stop` and then return the result of server.start/stop.